lib/jxl/enc_frame.cc

```diff
@@ -1173,6 +1372,25 @@ Status EncodeFrame(const CompressParams& cparams_orig,
                           ib.alpha(), lossless);
       }
     }
+    if (ApplyOverride(cparams.separate,
+                      cparams.speed_tier <= SpeedTier::kWombat &&
+                          cparams.resampling == 1 &&
+                          frame_header->encoding == FrameEncoding::kVarDCT)) {
+      JXL_ASSERT(cparams.resampling == 1);
+      ImageU separation(DivCeil(opsin.xsize(), kSeparationRes),
+                        DivCeil(opsin.ysize(), kSeparationRes));
+      float nonphoto = FindSeparation(opsin, &separation, cparams, pool);
+      if (WantDebugOutput(aux_out)) {
+        aux_out->DumpImage("separation", separation);
+      }
+      // TODO(jon): if nonphoto > 0.9, also try lossless
+      if (nonphoto > 0.2f) {
+        EncodeAndSubtract(&opsin, separation, cparams, passes_enc_state, pool,
+                          aux_out);
+        // TODO(jon): figure out why patches are broken
+        passes_enc_state->cparams.patches = Override::kOff;
```

Comment:
Maybe because they use the same reference frame index?

Reply:
Patches use index 0, this uses index 2, so that shouldn't be it, unless I am missing something?

Reply:
Fair enough. No clue then...